spec/models/chargeback_configured_system_spec.rb

```diff
@@ -0,0 +1,111 @@
+  end
+  let(:hourly_rate)               { 0.01 }
+  let(:count_hourly_rate)         { 1.00 }
+  let(:cpu_count)                 { 1.0 }
+  let(:memory_available)          { 1000.0 }
+  let(:starting_date) { Time.parse('2012-09-01 23:59:59Z').utc }
+  let(:ts) { starting_date.in_time_zone(Metric::Helper.get_time_zone(base_options[:ext_options])) }
+  let(:report_run_time) { month_end }
+  let(:month_beginning) { ts.beginning_of_month.utc }
+  let(:month_end) { ts.end_of_month.utc }
+  let(:hours_in_month) { Time.days_in_month(month_beginning.month, month_beginning.year) * 24 }
+  let(:ems) { FactoryBot.create(:ems_vmware) }
+
+  let(:hourly_variable_tier_rate)       { {:variable_rate => hourly_rate.to_s} }
+  let(:count_hourly_variable_tier_rate) { {:variable_rate => count_hourly_rate.to_s} }
+  let(:fixed_compute_tier_rate)         { {:fixed_rate => hourly_rate.to_s} }
+
+  let(:detail_params) do
+    {
+      :chargeback_rate_detail_cpu_allocated      => {:tiers => [count_hourly_variable_tier_rate]},
+      :chargeback_rate_detail_memory_allocated   => {:tiers => [hourly_variable_tier_rate]},
+      :chargeback_rate_detail_fixed_compute_cost => {:tiers => [fixed_compute_tier_rate]}
+    }
+  end
+
+  let!(:chargeback_rate) do
+    FactoryBot.create(:chargeback_rate, :detail_params => detail_params)
+  end
+
+  before do
+    MiqRegion.seed
+    ChargebackRateDetailMeasure.seed
+    ChargeableField.seed
+    ManageIQ::Showback::InputMeasure.seed
+    MiqEnterprise.seed
+
+    EvmSpecHelper.create_guid_miq_server_zone
+    cat = FactoryBot.create(:classification, :description => "Environment", :name => "environment", :single_value => true, :show => true)
+    c = FactoryBot.create(:classification, :name => "prod", :description => "Production", :parent_id => cat.id)
+    @tag = Tag.find_by(:name => "/managed/environment/prod")
+    ChargebackRate.set_assignments(:compute, [{:cb_rate => chargeback_rate, :tag => [c, "vm"]}])
+
+    @cs = FactoryBot.create(:configured_system, :created_at => report_run_time - 1.day, :computer_system =>
+            FactoryBot.create(:computer_system,
+                              :hardware => FactoryBot.create(:hardware,
+                                                             :cpu_total_cores => cores,
+                                                             :memory_mb       => mem_mb)))
+    @cs.tag_with(@tag.name, :ns => '*')
+
+    @cs2 = FactoryBot.create(:configured_system, :created_at => report_run_time - 1.day, :computer_system =>
+             FactoryBot.create(:computer_system,
+                               :hardware => FactoryBot.create(:hardware,
+                                                              :cpu_total_cores => cores,
+                                                              :memory_mb       => mem_mb)))
+    @cs2.tag_with(@tag.name, :ns => '*')
+    Timecop.travel(report_run_time)
+  end
+
+  after do
+    Timecop.return
+  end
+
+  context 'without metric rollups' do
```

Comment:
I guess, we don't have metric rollups for configured system, right ?

Reply:
That's correct, we do not have metric rollups.